services/cloudfront/src/main/java/software/amazon/awssdk/services/cloudfront/CloudFrontUtilities.java

```diff
@@ -0,0 +1,753 @@
+                                                         Instant expirationDate,
+                                                         String ipRange) throws InvalidKeySpecException, IOException {
+        PrivateKey privateKey = loadPrivateKey(privateKeyFile);
+        String resourceUrl = generateResourceUrl(protocol, distributionDomain, resourcePath);
+        return getCookiesForCustomPolicy(resourceUrl, privateKey, keyPairId, activeDate, expirationDate, ipRange);
+    }
+
+    /**
+     * Returns signed cookies that provides tailored access to private content based on an access time window and an ip range.
+     * For more information, see <a href=
+     * "https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/private-content-setting-signed-cookie-custom-policy.html"
+     * >Setting signed cookies using a custom policy</a>.
+     *
+     * @param resourceUrl       The URL or path for resource within a distribution.
+     * @param privateKey        Your private key file. RSA private key (.der) are supported.
+     * @param keyPairId          Identifier of a public/private certificate keypair already configured in your Amazon Web
+     *                           Services account.
+     * @param activeDate        The date from which content can be accessed using the generated cookies.
+     * @param expirationDate    The expiration date till which content can be accessed using the generated cookies.
+     * @param ipRange           The allowed IP address range of the client making the GET request,
+     *                          in CIDR form (e.g. 192.168.0.1/24).
+     * @return The signed cookies.
+     */
+    public static SignedCookie getCookiesForCustomPolicy(String resourceUrl,
+                                                         PrivateKey privateKey,
+                                                         String keyPairId,
+                                                         Instant activeDate,
+                                                         Instant expirationDate,
+                                                         String ipRange) {
+        try {
+            String policy = buildCustomPolicy(resourceUrl, activeDate, expirationDate, ipRange);
+            byte[] signatureBytes = signWithSha1Rsa(policy.getBytes(UTF_8), privateKey);
+            String urlSafePolicy = makeStringUrlSafe(policy);
+            String urlSafeSignature = makeBytesUrlSafe(signatureBytes);
+            return SignedCookie.builder()
+                               .isCustom(true).keyPairId(keyPairId).signature(urlSafeSignature).policy(urlSafePolicy).build();
+        } catch (InvalidKeyException e) {
+            throw SdkClientException.create("Could not sign custom policy cookie", e);
+        }
+    }
+
+    /**
+     * Contains common cookies used by Amazon CloudFront.
+     */
+    @SdkPublicApi
+    public static final class SignedCookie {
```

Comment:
This is a big class, it's better to extract `SignedCookie` to it's own class file.

Reply:
created CloudFrontSignedCookie class